homeassistant/components/launch_library/sensor.py

```diff
@@ -39,39 +43,73 @@ async def async_setup_platform(
     async_add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
-    """Create the launch sensor."""
-    name = config[CONF_NAME]
-    session = async_get_clientsession(hass)
-    launches = PyLaunches(session)
-
-    async_add_entities([LaunchLibrarySensor(launches, name)], True)
-
-
-class LaunchLibrarySensor(SensorEntity):
-    """Representation of a launch_library Sensor."""
-
-    _attr_icon = "mdi:rocket"
-
-    def __init__(self, api: PyLaunches, name: str) -> None:
-        """Initialize the sensor."""
-        self.api = api
-        self._attr_name = name
-
-    async def async_update(self) -> None:
-        """Get the latest data."""
-        try:
-            launches = await self.api.upcoming_launches()
-        except PyLaunchesException as exception:
-            _LOGGER.error("Error getting data, %s", exception)
-            self._attr_available = False
-        else:
-            if next_launch := next((launch for launch in launches), None):
-                self._attr_available = True
-                self._attr_native_value = next_launch.name
-                self._attr_extra_state_attributes = {
-                    ATTR_LAUNCH_TIME: next_launch.net,
-                    ATTR_AGENCY: next_launch.launch_service_provider.name,
-                    ATTR_AGENCY_COUNTRY_CODE: next_launch.pad.location.country_code,
-                    ATTR_STREAM: next_launch.webcast_live,
-                    ATTR_ATTRIBUTION: ATTRIBUTION,
-                }
+    """Import Launch Library configuration from yaml."""
+    _LOGGER.warning(
+        "Loading Launch Library via platform setup is deprecated. Please remove it from your yaml configuration"
+    )
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data=config,
```

Comment:
In the old version we can set a name, but in this one with the config flow i remove this feature as I don't see a point for it anymore with the possibility to change the entity names in the UI, is this correct?

Reply:
That is fine as long as the current name is properly imported and used

Reply:
Please see my latest commit, it should now include a reuse of the name if the config is imported from YAML. I also simplified the entity creation. I have removed the base entity to just make it a simpler change for this PR. I will split it up again when i add other sensors in a following PR.